lib/auth_plug.ex

```diff
@@ -66,4 +74,21 @@ defmodule AuthPlug do
   def create_jwt_session(conn, claims) do
     AuthPlug.Token.create_jwt_session(conn, claims)
   end
+
+
+  @doc """
+  `logout/1` does exactly what you expect; logs the person out of your app.
+  recieves a `conn` (Plug.Conn) and unsets the session.
+  This is super-useful in testing as we can easily reset a session.
+  """
+  def logout(conn) do
+    # https://stackoverflow.com/questions/42325996/delete-assigns
+    conn = update_in(conn.assigns, &Map.drop(&1, [:jwt, :person]))
+
+    conn
+    |> delete_session(:jwt) # hexdocs.pm/plug/Plug.Conn.html#delete_session/2
+    |> clear_session() # hexdocs.pm/plug/Plug.Conn.html#clear_session/1
+    |> configure_session(drop: true) # stackoverflow.com/questions/30999176
+    |> resp(302, "logged out")
```

Comment:
I'm not sure if it's necessary in this case but we can also add the `halt` function call to make sure that no other plug pipeline are called after the response:
```elixir
|> halt()
```

Reply:
@SimonLab this is a _very_ good point/question. I thought about it when doing the initial implementation. But ultimately left it out because it might not be desirable in the _consumer_ app. i.e. this is one of the rare occasions where we don't want to make the decision _for_ the person using `auth_plug`. We want to let them govern the lifecycle of their own `Plug.Conn`. 

See: https://github.com/dwyl/auth_plug_example/pull/16/files#r746566587 
Where the example app renders the `loggedout.html` template after invoking `AuthPlug.logout/1`

Reply:
Thanks for the explanation, make sense :+1: